### ISSUE TITLE
ORMs using node-sql?

### ISSUE THREAD

**Issue by taoeffect** [brianc/node-sql#320]
I'm kinda ticked off that it's 2016 and not a single node ORM uses SQL correctly. None of them use prepared statements / parameterized queries.

The good news though is that at least 3 of them ([bookshelf](https://github.com/tgriesser/bookshelf/), [objection.js](https://github.com/Vincit/objection.js), [firenze](https://github.com/fahad19/firenze)) use a single query-building library called [knex](https://github.com/tgriesser/knex).

If node-sql (or a wrapper to node-sql) conformed to knex's API, then all 3 of those ORMs would suddenly "do SQL right".

**Comment by taoeffect** [comment]
Looks like [Waterline](https://github.com/balderdashy/waterline) should make this fairly simple to do! 😄 

If I manage to create a Waterline adapter for node-sql I'll be sure to update this issue. Note there's also a feature-fork of it called [Offshore](https://github.com/Atlantis-Software/offshore).

**Comment by brianc** [comment]
sounds good! thanks!

**Comment by taoeffect** [comment]
It turns out that Knex actually does do parameterized queries, so I ended up going with objection.js for the ORM. However, the good news is that objection.js is just a thin wrapper around Knex, and theoretically it would be fairly simple to create a node-sql version of it (if someone wanted to give that a try).

Closing this issue since I (unfortunately) don't have the time available to work on that.
